### ISSUE TITLE
ag failed if the file is non-UTF-8

### ISSUE THREAD

**Issue by c02y** [ggreer/the_silver_searcher#1204]
I have been using `ag` for years, but a moment ago, I just found that ag finds nothing if the file is non-UTF-8 (in my case, it is ISO-8859), but tools like `grep`, `ack` and `rg` all successfully work.

Check the screenshot.
![snapshot6](https://user-images.githubusercontent.com/2135996/35376275-3ecc103c-01e5-11e8-9665-ec17f771c67e.png)

And Since I've been using `ag` for years, I'm really shocked that such a mature tool like `ag` doesn't support non-UTF-8 file.

**Comment by k-takata** [comment]
This is duplicate of #466.
`is_binary` is specialized for UTF-8 and it might return wrong results for non-UTF-8 texts.
I think the check should be loosen, or an option to skip the binary check is needed.

**Comment by c02y** [comment]
Yes, I searched the issues page before this post, and just like the duplicate page you mentioned, those issues are posted 3 or 4 years ago , but it is still not fixed.

I posted this new issue just to show how simple/common/serious this problem is.

**Comment by JFLarvoire** [comment]
FYI, in the Windows version of ag that I maintain [there](https://github.com/JFLarvoire/the_silver_searcher), I've implemented support for multiple file encodings, and this avoids the problem you're reporting. So this is feasible.  
It is a strongly desirable feature for Windows, because Windows, contrary to Linux, uses multiple file encodings simultaneously. For example, for the USA version of Windows, it's UTF-8 and code page 1252 (= ISO-8859-1) for text files, and code page 437 for the stdio pipes.  
And the good news is that, contrary to what I feared at first, this multiple-encoding support has negligible effect on performance.  
Anyway, if you're interesting in back-porting this feature into the master version of ag, please contact me.
